lib/honeybadger.ex

```diff
@@ -161,7 +170,26 @@ defmodule Honeybadger do
       hostname: :inet.gethostname |> elem(1) |> List.to_string,
       origin: "https://api.honeybadger.io",
       project_root: System.cwd,
-      use_logger: true,
-      environment_name: nil]
+      use_logger: true]
+  end
+
+  defp get_and_set_environment_name!(config) do
+    case System.get_env("MIX_ENV") do
```

Comment:
The way I do it for Ruby/Rails is:

1. RACK_ENV or RAILS_ENV environment variable (default)
2. Disk configuration (honeybadger.yml) (overrides default)
3. HONEYBADGER_ENV environment variable (always overrides disk config)

So I *do* prefer the environment variable if it belongs to the gem, but I give the least priority to 3rd party config. Does that make sense?